libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwner.kt

```diff
@@ -34,36 +35,56 @@ internal class CarAppLifecycleOwner : LifecycleOwner {
-            if (activitiesCreated == 0 && lifecycleCreated > 0 && lifecycleForegrounded == 0) {
-                changeState(Lifecycle.State.STARTED)
+            if (createdChangingConfiguration > 0) {
+                createdChangingConfiguration--
+            } else {
+                lifecycleCreated++
+                logger.i(TAG, Message("LifecycleOwner ($owner) onCreate"))
+                if (activitiesCreated == 0 && lifecycleCreated > 0 && lifecycleForegrounded == 0) {
+                    changeState(Lifecycle.State.STARTED)
+                }
             }
         }
 
         override fun onStart(owner: LifecycleOwner) {
-            lifecycleForegrounded++
-            logger.i(TAG, Message("LifecycleOwner ($owner) onStart"))
-            if (activitiesCreated == 0 && lifecycleForegrounded > 0) {
-                changeState(Lifecycle.State.RESUMED)
+            if (foregroundedChangingConfiguration > 0) {
+                foregroundedChangingConfiguration--
+            } else {
+                lifecycleForegrounded++
+                logger.i(TAG, Message("LifecycleOwner ($owner) onStart"))
+                if (activitiesCreated == 0 && lifecycleForegrounded > 0) {
+                    changeState(Lifecycle.State.RESUMED)
+                }
             }
         }
 
         override fun onStop(owner: LifecycleOwner) {
-            lifecycleForegrounded--
-            logger.i(TAG, Message("LifecycleOwner ($owner) onStop"))
-            if (activitiesForegrounded == 0 && lifecycleForegrounded == 0) {
-                changeState(Lifecycle.State.STARTED)
+            if (owner.isChangingConfigurations()) {
+                foregroundedChangingConfiguration++
+            } else {
+                lifecycleForegrounded--
+                logger.i(TAG, Message("LifecycleOwner ($owner) onStop"))
+                if (activitiesForegrounded == 0 && lifecycleForegrounded == 0) {
+                    changeState(Lifecycle.State.STARTED)
+                }
             }
         }
 
         override fun onDestroy(owner: LifecycleOwner) {
-            lifecycleCreated--
-            logger.i(TAG, Message("LifecycleOwner ($owner) onDestroy"))
-            if (activitiesForegrounded == 0 && lifecycleForegrounded == 0) {
-                changeState(Lifecycle.State.CREATED)
+            if (owner.isChangingConfigurations()) {
+                createdChangingConfiguration++
+            } else {
+                lifecycleCreated--
+                logger.i(TAG, Message("LifecycleOwner ($owner) onDestroy"))
+                if (activitiesForegrounded == 0 && lifecycleForegrounded == 0) {
+                    changeState(Lifecycle.State.CREATED)
+                }
             }
         }
+
+        private fun LifecycleOwner.isChangingConfigurations(): Boolean =
+            (this is Activity && this.isChangingConfigurations) ||
+                (this is Fragment && this.activity?.isChangingConfigurations == true)
```

Comment:
We have to check that this Fragment doesn't have `retainInstance` enabled. Because if it does, then it won't be destroyed after configuration changes.

Reply:
Another note, is that retaining fragments is deprecated in favor of ViewModels. 
https://developer.android.com/reference/androidx/fragment/app/Fragment.html#setRetainInstance(boolean)

![Screen Shot 2021-12-09 at 1 36 27 PM](https://user-images.githubusercontent.com/3021882/145479490-880f08fd-bde9-4706-b0c4-5c29050f927b.png)

Reply:
Logs showing everything is working as expected even with retain fragments. Here are logs from an orientation change.

```
2021-12-09 14:04:23.180 /kyle_debug: DropInActiveGuidance onStop
2021-12-09 14:04:23.180 /kyle_debug: ActiveGuidanceFragment onStop retained: true
2021-12-09 14:04:23.230 /kyle_debug: ActiveGuidanceFragment onDestroyView retained: true
2021-12-09 14:04:23.231 /kyle_debug: ActiveGuidanceFragment onDetach retained: true
2021-12-09 14:04:23.251 /kyle_debug: ActiveGuidanceFragment onAttach retained: true
2021-12-09 14:04:23.307 /kyle_debug: ActiveGuidanceFragment onCreateView retained: true
2021-12-09 14:04:23.332 /kyle_debug: ActiveGuidanceFragment onViewCreated retained: true
2021-12-09 14:04:23.334 /kyle_debug: ActiveGuidanceFragment onStart retained: true
2021-12-09 14:04:23.334 /kyle_debug: DropInActiveGuidance onStart
```

Ok I was running into some memory leaks when using a retain fragment. But, they're fixed now. I'm going to organize our example Fragment such that, it works when retaining fragments without memory leaks.

I've also verified that we do not need the `CarAppLifecycleOwner` to check `retainInstance` because it checks for "created" or "foregrounded" configuration changes. https://github.com/mapbox/mapbox-navigation-android/pull/5234#discussion_r766169328